src/boot/bootctl-uki.c

```diff
@@ -30,11 +30,16 @@ static int pe_sections(FILE *uki, struct PeSectionHeader **ret, size_t *ret_n) {
         assert(ret);
         assert(ret_n);
 
+        items = fread(&dos, 1, sizeof(dos.Magic), uki);
+        if (items != sizeof(dos.Magic))
+                return log_error_errno(SYNTHETIC_ERRNO(EIO), "DOS magic read error");
+        if (memcmp(dos.Magic, dos_file_magic, sizeof(dos_file_magic)) != 0)
+                goto no_sections;
+
+        rewind(uki);
         items = fread(&dos, 1, sizeof(dos), uki);
         if (items != sizeof(dos))
                 return log_error_errno(SYNTHETIC_ERRNO(EIO), "DOS header read error");
```

Comment:
Shouldn't those two just say `"File is smaller than DOS magic | DOS header (%zu bytes)", items`? "Read error" is misleading for the user, there is no "read error" here.